### PR TITLE
Add pre-commit linting of template files

### PR DESCRIPTION
**Pull request by jmsmkn**
With many developers now working on the project I've noticed that the template files are starting to get some errors. This PR adds `curlylint` as a pre-commit hook, which doesn't fix template errors but at least highlights them. Also fixes a few missing closing tags, indentation fixes will follow probably in another PR.

### CODE REVIEW

File: app/grandchallenge/algorithms/templates/algorithms/job_list_row.html [hunk 1]
```diff
@@ -11,13 +11,13 @@
    title="View result details">
     <i class="fa fa-info-circle"></i>
 </a>
-<split/>
+<split></split>
```
Comment (MikeOverkamp-diag) [inline app/grandchallenge/algorithms/templates/algorithms/job_list_row.html:14]: I think we need to revert these changes and all similar ones in the other templates. Or we need to change the processing here https://github.com/comic/grand-challenge.org/blob/master/app/grandchallenge/datatables/views.py#L30, which may be a better option.

**Review by MikeOverkamp-diag — CHANGES_REQUESTED** [review]
Just one comment.

File: app/grandchallenge/algorithms/templates/algorithms/job_list_row.html (hunk 1 above)
Comment (jmsmkn) [reply app/grandchallenge/algorithms/templates/algorithms/job_list_row.html:14]: The change to `datatables/views.py` is included in this PR.

Comment (MikeOverkamp-diag) [reply app/grandchallenge/algorithms/templates/algorithms/job_list_row.html:14]: Wow, sorry, I somehow completely missed that.

**Review by MikeOverkamp-diag — APPROVED** [review]

File: app/grandchallenge/algorithms/templates/algorithms/job_list_row.html (hunk 1 above)
Comment (jmsmkn) [reply app/grandchallenge/algorithms/templates/algorithms/job_list_row.html:14]: That's ok! The line is also covered well by tests.
